providers/src/airflow/providers/standard/utils/python_virtualenv.py

```diff
@@ -88,22 +145,33 @@ def prepare_virtualenv(
     if pip_install_options is None:
         pip_install_options = []
 
+    if requirements is not None and requirements_file_path is not None:
+        raise ValueError("Either requirements OR requirements_file_path has to be passed, but not both")
+
     if index_urls is not None:
         _generate_pip_conf(Path(venv_directory) / "pip.conf", index_urls)
 
-    venv_cmd = _generate_venv_cmd(venv_directory, python_bin, system_site_packages)
+    if _use_uv():
+        venv_cmd = _generate_uv_cmd(venv_directory, python_bin, system_site_packages)
+    else:
+        venv_cmd = _generate_venv_cmd(venv_directory, python_bin, system_site_packages)
     execute_in_subprocess(venv_cmd)
 
-    if requirements is not None and requirements_file_path is not None:
-        raise ValueError("Either requirements OR requirements_file_path has to be passed, but not both")
-
     pip_cmd = None
     if requirements is not None and len(requirements) != 0:
-        pip_cmd = _generate_pip_install_cmd_from_list(venv_directory, requirements, pip_install_options)
+        if _use_uv():
+            pip_cmd = _generate_uv_install_cmd_from_list(venv_directory, requirements, pip_install_options)
+        else:
+            pip_cmd = _generate_pip_install_cmd_from_list(venv_directory, requirements, pip_install_options)
     if requirements_file_path is not None and requirements_file_path:
```

Comment:
```suggestion
    else:
```
Line: 149 already checks if both requirement/requirement_file_path provided it raise exception, so in this case here else block is fine :) ?

Reply:
I also scratched my head regarding this... but the logic can be... use a venv without requirements file and without specified requirements... will install a "plain vanilla" runtime w/o any extra packages. There are even pytests for this. Seems to be a valid requirement having a "minimal with no sugar" venv to execute something within.

Reply:
Oh got it :) , thanks jens explaining.

Reply:
I also learned this while adding the feature :-D Something new every day :-D

Reply:
Do you think it would be worthwhile to add comments explaining why those conditions are set up that way? Otherwise, someone might come along later, think they're redundant, and try to remove them?

Reply:
If somebody cleans this up... pytests will fail.